core/os/time.cpp

```diff
@@ -124,50 +124,54 @@ VARIANT_ENUM_CAST(Time::Weekday);
 		}                                                                           \
 	}
 
-#define PARSE_ISO8601_STRING                                      \
-	int64_t year = UNIX_EPOCH_YEAR_AD;                            \
-	Month month = MONTH_JANUARY;                                  \
-	uint8_t day = 1;                                              \
-	uint8_t hour = 0;                                             \
-	uint8_t minute = 0;                                           \
-	uint8_t second = 0;                                           \
-	{                                                             \
-		bool has_date = false, has_time = false;                  \
-		String date, time;                                        \
-		if (p_datetime.find_char('T') > 0) {                      \
-			has_date = has_time = true;                           \
-			PackedStringArray array = p_datetime.split("T");      \
-			date = array[0];                                      \
-			time = array[1];                                      \
-		} else if (p_datetime.find_char(' ') > 0) {               \
-			has_date = has_time = true;                           \
-			PackedStringArray array = p_datetime.split(" ");      \
-			date = array[0];                                      \
-			time = array[1];                                      \
-		} else if (p_datetime.find_char('-', 1) > 0) {            \
-			has_date = true;                                      \
-			date = p_datetime;                                    \
-		} else if (p_datetime.find_char(':') > 0) {               \
-			has_time = true;                                      \
-			time = p_datetime;                                    \
-		}                                                         \
-		/* Set the variables from the contents of the string. */  \
-		if (has_date) {                                           \
-			PackedInt32Array array = date.split_ints("-", false); \
-			year = array[0];                                      \
-			month = (Month)array[1];                              \
-			day = array[2];                                       \
-			/* Handle negative years. */                          \
-			if (p_datetime.find_char('-') == 0) {                 \
-				year *= -1;                                       \
-			}                                                     \
-		}                                                         \
-		if (has_time) {                                           \
-			PackedInt32Array array = time.split_ints(":", false); \
-			hour = array[0];                                      \
-			minute = array[1];                                    \
-			second = array[2];                                    \
-		}                                                         \
+#define PARSE_ISO8601_STRING(err_val)                                                            \
```

Comment:
A bit of an inconsistent style with #60203 which uses `ret` as the argument name.